### PR TITLE
[systemtest][kafkabridge] Remove dependency of external listeners

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Enhancement 

### Description

This PR is focused on removing dependency of external listeners from Bridge tests. After discussion, we decided to use example-clients for sending/receiving messages via bridge service.

For our tests with weird usernames I had to keep the external listeners configuration -> so I moved it to separate suite. Earlier we found issue in our configuration -> the internal clients cannot handle names we used in weird usernames tests -> so with special characters, longer than (I think) 64 characters etc. I want to look at it in future PR, so I'll remove this suite later if it will be not needed.

I removed all of our HTTP methods, because we are using `example-http-clients` everywhere, so there is no reason to keep them. After I removed all the `WebClient` and `Vertx` usage, I had to remove it from `systemtests/pom.xml` -> the `mvn build` was failing because of unused dependency.

The change of `internalKafkaClient` properties (the change from `--broker-list` to `--bootstrap-servers`) has been done because of impossibility to send messages to `KafkaBridge` -> this problem was discussed with @ppatierno.

I added deployment of NP for Bridge -> this is needed for all operations between Bridge and other resources -> OCP4.x

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

File: systemtest/src/main/java/io/strimzi/systemtest/resources/KubernetesResource.java [hunk 1]
```diff
@@ -255,6 +263,42 @@ public static DoneableClusterRoleBinding clusterRoleBinding(ClusterRoleBinding c
...
+        backend.setServiceName(serviceName);
+        backend.setServicePort(new IntOrString(port));
+
+        HTTPIngressPath path = new HTTPIngressPath();
+        path.setPath("/");
+        path.setBackend(backend);
+
+        return new IngressBuilder()
+            .withNewMetadata()
+            .withName(serviceName)
```
Comment (ppatierno) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/KubernetesResource.java:277]: indentation

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeCors.java [hunk 2]
```diff
@@ -59,7 +53,7 @@ void testCorsOriginAllowed(VertxTestContext context) {
         JsonObject topics = new JsonObject();
         topics.put("topics", topic);
 
-        client.request(HttpMethod.OPTIONS, bridgePort, bridgeHost, "/consumers/" + groupId + "/instances/" + kafkaBridgeUser + "/subscription")
+        client.request(HttpMethod.OPTIONS, bridgeHost, "/consumers/" + groupId + "/instances/" + kafkaBridgeUser + "/subscription")
```
Comment (ppatierno) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeCors.java:56]: why the choice of makingthe host an explicit parameter while hiding the port (80) inside the method?

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeCors.java:56]: Yeah sorry, I leave that after some try, removed :)

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeExternalListenersST.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@Tag(NODEPORT_SUPPORTED)
+@Tag(EXTERNAL_CLIENTS_USED)
+class HttpBridgeExternalListenersST extends HttpBridgeBaseST {
+    private static final String BRIDGE_EXTERNAL_SERVICE = CLUSTER_NAME + "-bridge-external-service";
+
+    @Test
+    void testScramShaAuthWithWeirdNamedUser() throws Exception {
```
[the remark this answers is not in the document]
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeExternalListenersST.java:44]: Which problematic characters do you mean? Like `*+/` etc? I don't know if we didn't discussed that with @scholzj, but I can try it :)

[the remark this answers is not in the document]
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeExternalListenersST.java:44]: I tried some options and it's not working with these special characters, how I said, I think that we had conversation about the naming with @scholzj when we made the tests, so I think we could leave it like this.

Do you agree @Frawless, @samuel-hawker, @scholzj?

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=HttpBridgeST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Comment by Frawless** [comment]
Or we can simply send it via curl from some pod.

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Comment by im-konge** [comment]
@strimzi-ci run tests crc test_only profile=bridge

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Comment by im-konge** [comment]
@ppatierno @scholzj @samuel-hawker @Frawless @see-quick 

This is ready for review, but I know I'm going to change some things. I need to test this on OCP4.x to be 100% everything works as expected. Thanks a lot.

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeAbstractST.java [hunk 4]
```diff
@@ -37,40 +27,18 @@
 @Tag(NODEPORT_SUPPORTED)
 @Tag(EXTERNAL_CLIENTS_USED)
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeAbstractST.java:28]: Why do we have these tags if we are removing external access?

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeCorsST.java [hunk 5]
```diff
@@ -4,129 +4,109 @@
...
         KafkaBridgeHttpCors kafkaBridgeHttpCors = KafkaBridgeResource.kafkaBridgeClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getSpec().getHttp().getCors();
         LOGGER.info("Bridge with the following CORS settings {}", kafkaBridgeHttpCors.toString());
 
-        Service service = KafkaBridgeUtils.createBridgeNodePortService(CLUSTER_NAME, NAMESPACE, bridgeExternalService);
-        KubernetesResource.createServiceResource(service, NAMESPACE).done();
-        ServiceUtils.waitForNodePortService(bridgeExternalService);
-
-        bridgePort = KafkaBridgeUtils.getBridgeNodePort(NAMESPACE, bridgeExternalService);
-        bridgeHost = kubeClient(NAMESPACE).getNodeAddress();
+        podName = kubeClient().listPodsByPrefixInName(CLUSTER_NAME + "-bridge").get(0).getMetadata().getName();
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeCorsST.java:110]: I think you should be running this in a separate pod. Not from the bridge pod it self.

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeAbstractST.java (hunk 4 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeAbstractST.java:28]: I forgot to remove it, thanks

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeCorsST.java (hunk 5 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeCorsST.java:110]: I'm gonna take a look at it :)

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java [hunk 6]
```diff
@@ -44,152 +48,181 @@ public static JsonObject generateHttpMessages(int messageCount) {
...
+                    String key = jsonObject.getString("key");
+                    Object value = jsonObject.getValue("value");
+                    long offset = jsonObject.getLong("offset");
+                    LOGGER.debug("Received msg: topic:{} partition:{} key:{} value:{} offset{}", kafkaTopic, kafkaPartition, key, value, offset);
                 }
-            });
-        return future.get(1, TimeUnit.MINUTES);
+                LOGGER.info("Received {} messages from KafkaBridge", jsonResponse.size());
+            } else {
+                LOGGER.warn("Received body 0 messages: {}", jsonResponse);
```
[the remark this answers is not in the document]
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java]: already changed -> these methods are useless right now as we are using everywhere example clients...

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java [hunk 7]
```diff
@@ -44,152 +48,181 @@ public static JsonObject generateHttpMessages(int messageCount) {
...
+        if (jsonResponse.size() == 0) {
+            LOGGER.info("Received 0 messages, trying again after subscribing to offset");
+            jsonResponse = receiveMessages(podName, groupID, name);
+        }
+
+        return jsonResponse;
+    }
+
+    public static JsonArray receiveMessages(String podName, String groupID, String name) {
+        LOGGER.info("Receiving records from KafkaBridge");
```
[the remark this answers is not in the document]
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java]: same as above

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java [hunk 8]
```diff
@@ -44,152 +48,181 @@ public static JsonObject generateHttpMessages(int messageCount) {
         return root;
     }
 
-    public static JsonObject sendMessagesHttpRequest(JsonObject records, String bridgeHost, int bridgePort, String topicName, WebClient client) throws InterruptedException, ExecutionException, TimeoutException {
+    public static JsonObject sendMessagesHttpRequest(JsonObject records, String topicName, String podName) {
         LOGGER.info("Sending records to KafkaBridge");
```
[the remark this answers is not in the document]
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java]: same as above

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java [hunk 9]
```diff
@@ -4,31 +4,35 @@
...
+import static io.strimzi.systemtest.resources.ResourceManager.cmdKubeClient;
 
 public class BridgeUtils {
 
     private static final Logger LOGGER = LogManager.getLogger(HttpUtils.class);
 
+    public static final Pattern ALL_BEFORE_JSON_PATTERN = Pattern.compile("(.*\\s)\\{", Pattern.DOTALL);
+    private static final Pattern ALL_BEFORE_JSON_ARRAY_PATTERN = Pattern.compile("(.*\\s)\\[", Pattern.DOTALL);
+
+    public static final String DEFAULT_BRIDGE_HOST = "localhost:" + Constants.HTTP_BRIDGE_DEFAULT_PORT;
```
[the remark this answers is not in the document]
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java]: already changed

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=oauth

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Comment by im-konge** [comment]
@scholzj @samuel-hawker @ppatierno @Frawless @see-quick 

This is ready for review now. Everything is fixed for `OCP4.x`

File: systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java [hunk 10]
```diff
@@ -62,7 +62,7 @@ public static void waitForClientSuccess(String jobName, String namespace, int me
 
     private static long timeoutForClientFinishJob(int messagesCount) {
         // need to add at least 1-2minutes for finishing the job
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java:64]: Change commend ? :)

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java [hunk 11]
```diff
@@ -4,192 +4,76 @@
...
     private BridgeUtils() { }
 
-    public static JsonObject generateHttpMessages(int messageCount) {
-        LOGGER.info("Creating {} records for KafkaBridge", messageCount);
-        JsonArray records = new JsonArray();
-        JsonObject json = new JsonObject();
-        for (int i = 0; i < messageCount; i++) {
-            json.put("value", "msg_" + i);
-            records.add(json);
+    public static String getCurlCommand(HttpMethod httpMethod, String url, String headers, String data) {
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java:26]: I would prefer called it `buildCurlCommand` but it's just  suggestion.

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java [hunk 12]
```diff
@@ -4,192 +4,76 @@
...
-                if (ar.succeeded() && ar.result().statusCode() == 204) {
-                    LOGGER.info("Consumer subscribed");
-                    future.complete(ar.succeeded());
-                } else {
-                    LOGGER.error("Cannot subscribe consumer", ar.cause());
-                    future.completeExceptionally(ar.cause());
-                }
-            });
-        return future.get(1, TimeUnit.MINUTES);
+    public static String executeCurlCommand(HttpMethod httpMethod, String podName, String data, String url, String headers) {
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java:40]: I know that the base idea was good. In a practise if you create method, which has one line to encapsulate some logic. You can end-up having these `one-liners` with different names but most likely same behaviour. Wouldn't you think that having:
```
cmdKubeClient().execInPod(podName, "/bin/bash", "-c", BridgeUtils.getCurlCommand(httpMethod, url, headers, data)).out().trim();
```
will be better?

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java [hunk 13]
```diff
@@ -4,225 +4,115 @@
...
             .withNamespaceName(NAMESPACE)
             .withClusterName(CLUSTER_NAME)
+            .withMessageCount(MESSAGE_COUNT)
             .withKafkaUsername(USER_NAME)
-            .withMessageCount(messageCount)
-            .withSecurityProtocol(SecurityProtocol.SASL_SSL)
+            .withUsingPodName(kafkaClientsPodName)
             .build();
 
-        assertThat(kafkaClient.receiveMessagesTls(), is(messageCount));
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java:75]: The point of this test was to send the encrypted (tls) and moreover with support of simple authentication (scram-sha). You have changed to plain communication. Why? I am assuming that you wanted to just test thee scram-sha with plain communication. If this is the case please change the name of the test :)

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java [hunk 14]
```diff
@@ -4,191 +4,102 @@
...
-import static io.strimzi.systemtest.Constants.EXTERNAL_CLIENTS_USED;
-import static io.strimzi.systemtest.Constants.NODEPORT_SUPPORTED;
-import static io.strimzi.systemtest.Constants.REGRESSION;
-import static io.strimzi.systemtest.bridge.HttpBridgeST.NAMESPACE;
+import static io.strimzi.systemtest.Constants.INTERNAL_CLIENTS_USED;
 import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
 
-@Tag(BRIDGE)
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java:44]: Why do you remove BRIDGE + REGRESSION tag?

File: systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java (hunk 10 above)
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java:64]: :+1:

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java (hunk 14 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java:44]: It takes all tags from `HttpAbstractST`, so adding it here is not necessary.

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java (hunk 13 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java:75]: My bad! I will change it :) thanks

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java [hunk 15]
```diff
@@ -4,192 +4,76 @@
...
 
-    public static JsonObject createBridgeConsumer(JsonObject config, String bridgeHost, int bridgePort, String groupId,
-                                                  WebClient client, Map<String, String> additionalHeaders) throws InterruptedException, ExecutionException, TimeoutException {
+    public static String addHeadersToString(Map<String, String> additionalHeaders,  String contentType, String content) {
+        StringBuilder headerString = new StringBuilder();
 
-        MultiMap headers = MultiMap.caseInsensitiveMultiMap()
-            .add("Content-length", String.valueOf(config.toBuffer().length()))
-            .add("Content-type", Constants.KAFKA_BRIDGE_JSON);
+        if (!content.equals("")) {
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java:51]: Same as above

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java [hunk 16]
```diff
@@ -4,192 +4,76 @@
...
+            headerString.append(" -H 'Content-length: ").append(content.length()).append("'");
+        }
+
+        if (!contentType.equals("")) {
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java:55]: same as above

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeAbstractST.java [hunk 17]
```diff
@@ -34,43 +22,19 @@
...
-@Tag(EXTERNAL_CLIENTS_USED)
 public class HttpBridgeAbstractST extends AbstractST {
-    private static final Logger LOGGER = LogManager.getLogger(HttpBridgeAbstractST.class);
-
-    protected WebClient client;
-    protected String bridgeExternalService = CLUSTER_NAME + "-bridge-external-service";
+    public static int bridgePort = Constants.HTTP_BRIDGE_DEFAULT_PORT;
+    public static String bridgeHost = "";
+    public static String kafkaClientsPodName = "";
+    public static String bridgeServiceName = KafkaBridgeResources.serviceName(CLUSTER_NAME);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeAbstractST.java:29]: Wouldn't be better to use `KafkaBridgeResources.url()` instead just service name?

File: systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java [hunk 18]
```diff
@@ -62,7 +62,7 @@ public static void waitForClientSuccess(String jobName, String namespace, int me
...
-        return (long) messagesCount * 1000 + Duration.ofMinutes(2).toMillis();
+        return (long) messagesCount * 1000 + Duration.ofMinutes(3).toMillis();
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java:65]: Why?

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeExternalListenersST.java [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+import org.junit.jupiter.api.Test;
+
+import static io.strimzi.systemtest.Constants.EXTERNAL_CLIENTS_USED;
+import static io.strimzi.systemtest.Constants.NODEPORT_SUPPORTED;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@Tag(NODEPORT_SUPPORTED)
+@Tag(EXTERNAL_CLIENTS_USED)
+class HttpBridgeExternalListenersST extends HttpBridgeAbstractST {
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeExternalListenersST.java:39]: The name is kinda misleading from my POV. IT looks like bridge have external listener here, which is not true.

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java [hunk 20]
```diff
@@ -4,225 +4,115 @@
...
-        // Try to consume messages
-        JsonArray bridgeResponse = BridgeUtils.receiveMessagesHttpRequest(bridgeHost, bridgePort, CONSUMER_GROUP_NAME, aliceUser, client);
-        if (bridgeResponse.size() == 0) {
-            // Real consuming
-            bridgeResponse = BridgeUtils.receiveMessagesHttpRequest(bridgeHost, bridgePort, CONSUMER_GROUP_NAME, aliceUser, client);
-        }
-        assertThat("Sent message count is not equal with received message count", bridgeResponse.size(), is(MESSAGE_COUNT));
-        // Delete consumer
-        assertThat(BridgeUtils.deleteConsumer(bridgeHost, bridgePort, CONSUMER_GROUP_NAME, aliceUser, client), is(true));
+        assertThat(internalKafkaClient.sendMessagesPlain(), is(MESSAGE_COUNT));
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java:73]: Same as Maros mentioned above I guess

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java [hunk 21]
```diff
@@ -4,225 +4,115 @@
...
         // Deploy kafka
         KafkaResource.kafkaEphemeral(CLUSTER_NAME, 1, 1)
             .editSpec()
                 .editKafka()
                     .withNewListeners()
-                        .withNewKafkaListenerExternalNodePort()
-                            .withAuth(new KafkaListenerAuthenticationScramSha512())
-                        .endKafkaListenerExternalNodePort()
-                        .withNewTls().withAuth(new KafkaListenerAuthenticationScramSha512()).endTls()
+                        .withNewPlain().withAuth(new KafkaListenerAuthenticationScramSha512()).endPlain()
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java:88]: Some new lines/indents?

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java [hunk 22]
```diff
@@ -4,225 +4,115 @@
...
         passwordSecret.setPassword("password");
 
-        // Initialize CertSecretSource with certificate and secret names for consumer
-        CertSecretSource certSecret = new CertSecretSource();
-        certSecret.setCertificate("ca.crt");
-        certSecret.setSecretName(KafkaResources.clusterCaCertificateSecretName(CLUSTER_NAME));
-
         // Deploy http bridge
-        KafkaBridgeResource.kafkaBridge(CLUSTER_NAME, KafkaResources.tlsBootstrapAddress(CLUSTER_NAME), 1)
+        KafkaBridgeResource.kafkaBridge(CLUSTER_NAME, KafkaResources.plainBootstrapAddress(CLUSTER_NAME), 1)
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeScramShaST.java:106]: Probably same as above? tls -> plain

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java (hunk 14 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java:44]: I wonder if we should keep it in all classes instead of inheriting it. It's a little bit confusing.

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java [hunk 23]
```diff
@@ -4,191 +4,102 @@
...
         // Deploy kafka
         KafkaResource.kafkaEphemeral(CLUSTER_NAME, 1, 1)
             .editSpec()
                 .editKafka()
-                    .editListeners()
-                        .withNewKafkaListenerExternalNodePort()
-                            .withAuth(new KafkaListenerAuthenticationTls())
-                        .endKafkaListenerExternalNodePort()
+                    .withNewListeners()
+                        .withNewTls().withAuth(new KafkaListenerAuthenticationTls()).endTls()
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java:91]: I think format before this changes looks better.

File: systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java [hunk 24]
```diff
@@ -556,7 +551,7 @@ void testProducerConsumerMirrorMakerService() {
         KafkaClientsResource.consumerWithTracing(KafkaResources.plainBootstrapAddress(kafkaClusterTargetName)).done();
 
         KafkaMirrorMakerResource.kafkaMirrorMaker(CLUSTER_NAME, kafkaClusterSourceName, kafkaClusterTargetName,
-            ClientUtils.generateRandomConsumerGroup(), 1, false)
+                "my-group" + new Random().nextInt(Integer.MAX_VALUE), 1, false)
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java:554]: Why this change is needed?

File: systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java [hunk 25]
```diff
@@ -688,7 +683,7 @@ void testProducerConsumerMirrorMakerConnectStreamsService() {
                 + "'" + connectorConfig + "'" + " http://localhost:8083/connectors");
 
         KafkaMirrorMakerResource.kafkaMirrorMaker(CLUSTER_NAME, kafkaClusterSourceName, kafkaClusterTargetName,
-            ClientUtils.generateRandomConsumerGroup(), 1, false)
+                "my-group" + new Random().nextInt(Integer.MAX_VALUE), 1, false)
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java:686]: Same as above

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/ClientArgument.java [hunk 26]
```diff
@@ -23,7 +23,7 @@
     ASSIGMENT_STRATEGY("--assignment-strategy"),
 
     // Producer
-    BROKER_LIST("--broker-list"),
+    BOOTSTRAP_SERVER("--bootstrap-server"),
```
[the remark this answers is not in the document]
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/ClientArgument.java:26]: Yes it should, gonna test it on ocp4.x too

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java [hunk 27]
```diff
@@ -4,192 +4,76 @@
...
-        LOGGER.info("Creating {} records for KafkaBridge", messageCount);
-        JsonArray records = new JsonArray();
-        JsonObject json = new JsonObject();
-        for (int i = 0; i < messageCount; i++) {
-            json.put("value", "msg_" + i);
-            records.add(json);
+    public static String getCurlCommand(HttpMethod httpMethod, String url, String headers, String data) {
+        String command = "curl -X " + httpMethod.toString() + " -D - " + url + " " + headers;
+
+        if (!data.equals("")) {
```
[the remark this answers is not in the document]
Comment (see-quick) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java:29]: but with the `!` ofc :D

File: systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java (hunk 18 above)
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java:65]: In some cases the job needed one more minute on Jenkins -> on local env two minutes are sufficient ...

File: systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java (hunk 24 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java:554]: Good catch! I think I little bit messed up the rebase

File: systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java (hunk 25 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/tracing/TracingST.java:686]: Same as above ^^

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java (hunk 11 above)
Comment (ppatierno) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/specific/BridgeUtils.java:26]: I have another suggestion, but it's just that. You could also check that data has to be empty when the method is not  PUT or POST.

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeExternalListenersST.java (hunk 19 above)
Comment (ppatierno) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeExternalListenersST.java:39]: Agree

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java (hunk 14 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeTlsST.java:44]: I'm gonna add it back and remove the inheritance.

File: systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeAbstractST.java (hunk 17 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeAbstractST.java:29]: For this case I'm gonna add `bridgeUrl` var to use `KafkaBridgeResources.url()` method and I will keep `bridgeServiceName` var for example clients usage.

File: systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/ClientArgument.java (hunk 26 above)
[the remark this answers is not in the document]
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/internalClients/ClientArgument.java:26]: It works

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

File: systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java (hunk 18 above)
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java:65]: I think that 2 minutes will be enough -> I forgot that I added the `auto.reset.offset.config: earliest` so I'm gonna change it back

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Review by scholzj — APPROVED** [review]

**Review by see-quick — APPROVED** [review]

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=bridge

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=oauth
